src/main/java/bio/terra/common/LogPrintable.java

```diff
@@ -0,0 +1,10 @@
+package bio.terra.common;
+
+/**
+ * Interface created to have a consistent name for a method to print values to logs and error
+ * messages
+ */
+public interface LogPrintable {
```

Comment:
tpString often (and in this case) has more information than we want to log (e.g. tostring for a dataset would include schema, etc.). having an explicit method with the intended purpose of the output I found a little cleaner

Reply:
`toString()` for a dataset doesn't exist, and would only be used by us anyway, so it can include whatever we want it to. There's no reason why `toString()` would include any data we don't want it to.

To me, printing an object to the log is like printing it anywhere else, and `toString()` seems like the way to express that operation.

If at some point we did have a need to implement a longer/shorter version of `toString()`, we could think about how to express the different operations then.

Reply:
This is why I don't like depending on toString. There are different uses for it.  We straight up can't log all of the information for a dataset since it won't fit in log message limits from stackdriver (and regardless, would make debugging a nightmare).  This is really just to return the identifying information to allow someone debugging to decide what information is useful to look up from there

Reply:
> This is why I don't like depending on toString. There are different uses for it. We straight up can't log all of the information for a dataset since it won't fit in log message limits from stackdriver (and regardless, would make debugging a nightmare).

Maybe I'm not being clear about what I'm suggesting. In the classes where `LogPrintable` is used, there is no `toString()` method defined currently. So instead of adding a `toLogString()` method, I would write these methods as `toString()` using the current definitions of your `toLogString()` methods. The program behavior would be the same as it is currently in your PR; the main difference is that the calling code doesn't need to know that these objects have a special type which provides a printable version of the object (since that is what `toString()` is supposed to do anyway).